examples/basic/evm-example/src/components/SwapContent/SwapContent.tsx

```diff
@@ -0,0 +1,599 @@
+          `Invalid quote response: ${quote.resultType}, please try again.`
+        )
+        setLoadingSwap(false)
+      } else {
+        await executeRoute(
+          from,
+          to,
+          fromAddress,
+          toAddress,
+          amount
+          // sampleImMessage
+        )
+      }
+    } catch (error) {
+      setError(`Error requesting quote: ${error}`)
+      setLoadingSwap(false)
+    }
+  }
+
+  const executeRoute = async (
+    from: Asset,
+    to: Asset,
+    fromAddress: string,
+    toAddress: string,
+    inputAmount: string
+    // imMessage?: string
+  ) => {
+    if (!fromToken || !toToken || !fromChain || !toChain) return
+
+    const signer = getSigners(
+      getWalletByChainType(fromChain?.type as TransactionType)
+    ).getSigner(fromChain?.type as TransactionType)
+
+    let swap: SwapResponse | null = null
+
+    try {
+      let swapRequest: SwapRequest = {
+        from,
+        to,
+        amount: inputAmount,
+        fromAddress: fromAddress,
+        toAddress: toAddress,
+        disableEstimate: false,
+        referrerAddress: null,
+        referrerFee: null,
+        slippage: '1.0',
+        swapperGroups: disabledLiquiditySources,
+        swappersGroupsExclude: true,
+      }
+      if (testMessagePassing) {
+        swapRequest = {
+          ...swapRequest,
+          messagingProtocols: selectedProtocols,
+          sourceContract: '???', // TODO
+          destinationContract: '???', // TODO
+          // imMessage,
+        }
+      }
+      console.log({ swapRequest })
+      swap = await sdk.swap(swapRequest)
+      console.log({ swapResponse: swap })
+
+      if (!!swap.error || swap.resultType !== RoutingResultType.OK) {
+        setError(
+          `Error swapping, routing result: ${swap.resultType}, error: ${swap.error}`
+        )
+        setLoadingSwap(false)
+        return
+      }
+
+      if (fromChain.type === TransactionType.EVM) {
+        const tx = swap.tx as EvmTransaction
+
+        if (tx) {
+          // if approve data is not null, it means approve needed, otherwise it's already approved.
+          if (!!tx?.approveData) {
+            // try to approve
+
+            const result = await signer.signAndSendTx(
+              {
+                type: TransactionType.EVM,
+                blockChain: tx.blockChain?.name,
+                isApprovalTx: !!tx.approveData,
+                from: tx.from,
+                to: tx.txTo,
+                data: tx.txData,
+                value: tx.value,
+                nonce: null,
+                gasLimit: tx.gasLimit,
+                gasPrice: tx.gasPrice,
+                maxPriorityFeePerGas: tx.maxPriorityFeePerGas,
+                maxFeePerGas: tx.maxFeePerGas,
+              },
+              fromAddress,
+              fromChain.chainId
+            )
+            await checkApprovalSync(swap.requestId, result.hash, sdk)
+            console.log('transaction approved successfully')
+          } else {
```

Comment:
This else seems to be wrong. You need to sign the main transaction anyway even if we need approval.

Reply:
Thanks Fixed